InnerEye/ML/reports/classification_report.py

```diff
@@ -52,84 +54,220 @@ class ReportedMetrics(Enum):
                                                            labels_and_model_outputs.model_outputs)
-    plot_auc(recall, precision, "PR Curve", ax[1])
+    axs[1].plot(recall, precision, **plot_kwargs)
+    format_pr_or_roc_axes('pr', axs[1])
+
+    plt.show()
+
+
+def plot_scores_and_summary(all_labels_and_model_outputs: Sequence[LabelsAndPredictions],
+                            scoring_fn: Callable[[LabelsAndPredictions], Tuple[np.ndarray, np.ndarray]],
+                            confidence_interval_width: float = .8,
+                            ax: Optional[Axes] = None) -> Tuple[List, Any]:
+    """
+    Plot a collection of score curves along with the (vertical) median and confidence interval (CI).
+
+    Each plotted curve is interpolated onto a common horizontal grid, and the median and CI are computed vertically
+    at each horizontal location.
+    :param all_labels_and_model_outputs: Collection of ground-truth labels and model predictions (e.g. for various
+    cross-validation runs).
+    :param scoring_fn: A scoring function mapping a `LabelsAndPredictions` object to X and Y coordinates for plotting.
+    :param confidence_interval_width: A value in [0, 1] representing what fraction of the data should be contained in
+    the shaded area. The edges of the CI are `median +/- confidence_interval_width/2`.
+    :param ax: Axes object onto which to plot (default: use current axes).
+    :return: A tuple of `(line_handles, summary_handle)` to use in setting a legend for the plot: `line_handles` is a
+    list corresponding to the curves for each `LabelsAndPredictions`, and `summary_handle` references the median line
+    and shaded CI area.
+    """
+    if ax is None:
+        ax = plt.gca()
+    x_grid = np.linspace(0, 1, 101)
+    interp_ys = []
+    line_handles = []
+    for index, labels_and_model_outputs in enumerate(all_labels_and_model_outputs):
+        x_values, y_values = scoring_fn(labels_and_model_outputs)
+        interp_ys.append(np.interp(x_grid, x_values, y_values))
+        handle, = ax.plot(x_values, y_values, lw=1)
+        line_handles.append(handle)
+
+    confidence_interval_quantiles = [.5 - confidence_interval_width / 2, .5, .5 + confidence_interval_width / 2]
+    y_lo, y_mid, y_hi = quantile(interp_ys, confidence_interval_quantiles, axis=0)
+    h1 = ax.fill_between(x_grid, y_lo, y_hi, color='k', alpha=.2, lw=0)
+    h2, = ax.plot(x_grid, y_mid, 'k', lw=2)
+    summary_handle = (h1, h2)
+    return line_handles, summary_handle
+
+
+def plot_pr_and_roc_curves_crossval(all_labels_and_model_outputs: Sequence[LabelsAndPredictions],
+                                    axs: Optional[Sequence[Axes]] = None) -> None:
+    """
+    Given a list of LabelsAndPredictions objects, plot the corresponding ROC and PR curves, along with median line and
+    shaded 80% confidence interval (computed over TPRs and precisions for each fixed FPR and recall value).
```

Comment:
Why 80%?

Reply:
this choice was somewhat arbitrary. I thought 95% didn't make sense if we have only ~5–10 curves, and 50% (IQR) could be too narrow. but happy to take suggestions if people have a better idea